### PR TITLE
Feature - Custom Result Initializers

### PR DESCRIPTION
**Pull request by cnoon**
This PR adds custom initializers for the `Result` success and failure cases. It also adds tests around the usage.

### CODE REVIEW

**Comment by cnoon** [comment]
cc @kylef and @kcharwood

File: Source/Result.swift [hunk 1]
```diff
@@ -34,6 +34,29 @@ public enum Result<Value> {
...
     case Failure(NSData?, NSError)
 
+    /**
+        Creates a `.Success` result with the given value.
+
+        - parameter value: The value to store in the `.Success`.
+
+        - returns: A new `.Success` result.
+    */
+    public init(value: Value) {
```
Comment (kylef) [inline Source/Result.swift]: I'm wondering, do they need to be public? Should anyone outside of Alamofire be creating a Result?

Comment (cnoon) [inline Source/Result.swift]: That's really a good point. I was a bit torn as well. After adding the tests, I realized it's not any easier to use the initializers than the default case initialization. Maybe this whole PR should just be declined. What you think @kylef?

Comment (kylef) [inline Source/Result.swift]: It's up to you, but I'd only merge this if we want to use it in Alamofire.

Comment (cnoon) [inline Source/Result.swift]: Okay, I'm going to close it out. I don't think we'll end up using it.

**Pull request closed without merging**